model/role.go

```diff
@@ -484,11 +484,17 @@ func (r *Role) IsValidWithoutId() bool {
 
 	for _, permission := range r.Permissions {
 		permissionValidated := false
-		for _, p := range append(AllPermissions, DeprecatedPermissions...) {
-			if permission == p.Id {
-				permissionValidated = true
-				break
+		check := func(perms []*Permission, permission string) bool {
```

Comment:
Wondering if having the check function defined just outside the loop would make things even more readable.